### PR TITLE
Better message for creating session with W3C

### PR DESCRIPTION
**Pull request by Siddh010**
Create PR for [appium/appium/issues/12766]
Better message for creating session with W3C

### CODE REVIEW

File: lib/basedriver/commands/session.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import log from '../logger';
 import { errors } from '../../protocol';
 import UUID from 'uuid-js';
 import { processCapabilities } from '../capabilities';
+import prettyJson from 'prettyjson';
```
Comment (mykola-mokhnach) [inline lib/basedriver/commands/session.js]: what is the point in using third-party dependency? we can still add indentation parameters to JSON.stringify call

**Comment by Siddh010** [comment]
Please find the few different ways  I tried for making W3C capabilities more readable and showing in JSON format, let me know which look fine. 

1) Using JSON.stringify(w3cCapabilities) 
     log.debug(`Creating session with W3C capabilities: ${JSON.stringify(w3cCapabilities)}`);
  
   OutPut in Appium server:- 
         [BaseDriver] Creating session with W3C capabilities: {"alwaysMatch": 
             {"platformName":"android","appium:app":"/Users/siddartha/eclipse-workspace/AppiumSampleDemo/VodQA.apk","appium:automationName":"UIAutomator2","appium:deviceName":"android","appium:platformVersion":"8.1"},"firstMatch":[{}]}

2) Using prettyJson lib

      log.debug(`Creating session with W3C capabilities:`);
      log.debug(prettyJson.render(w3cCapabilities, {noColor: true}));

    Output in the appium server log

[debug] [BaseDriver] W3C capabilities and MJSONWP desired capabilities were provided
[debug] [BaseDriver] Creating session with W3C capabilities:
[debug] [BaseDriver] alwaysMatch: 
[debug] [BaseDriver]   platformName:           android
[debug] [BaseDriver]   appium:app:             /Users/siddartha/eclipse-workspace/AppiumSampleDemo/VodQA.apk
[debug] [BaseDriver]   appium:automationName:  UIAutomator2
[debug] [BaseDriver]   appium:deviceName:      android
[debug] [BaseDriver]   appium:platformVersion: 8.1
[debug] [BaseDriver] firstMatch: 
[debug] [BaseDriver]   -


3) Before: we were using 
 log.debug(`Creating session with W3C capabilities: ${_.truncate(JSON.stringify(w3cCapabilities))}`);

OutPut in Appium Server log:- 
[debug] [BaseDriver] W3C capabilities and MJSONWP desired capabilities were provided
[debug] [BaseDriver] Creating session with W3C capabilities: {"alwaysMatch":{"platformNa...

Also tried with inspectObject method from (https://github.com/appium/appium/blob/master/lib/utils.js) 
but showing empty/null  array object for firstMatch

[BaseDriver]   firstMatch: {
[BaseDriver]     0: {
[BaseDriver]     }
[BaseDriver]   }

Output 

[debug] [BaseDriver] W3C capabilities and MJSONWP desired capabilities were provided
[debug] [BaseDriver] Creating session with W3C capabilities:
[BaseDriver]   alwaysMatch: {
[BaseDriver]     platformName: android
[BaseDriver]     appium:app: /Users/siddartha/eclipse-workspace/AppiumSampleDemo/VodQA.apk
[BaseDriver]     appium:automationName: UIAutomator2
[BaseDriver]     appium:deviceName: android
[BaseDriver]     appium:platformVersion: 8.1
[BaseDriver]   }
[BaseDriver]   firstMatch: {
[BaseDriver]     0: {
[BaseDriver]     }
[BaseDriver]   }

**Comment by mykola-mokhnach** [comment]
Check the documentation on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify

The function also has the second and the third (`replacer, space`) optional arguments, which can be of use.

**Comment by Siddh010** [comment]
Thanks, Mykola, used  below code,  please let me know if this is fine 
log.debug(`Creating session with W3C capabilities: ${JSON.stringify(w3cCapabilities,null,'\t')}`);

Output:
[debug] [BaseDriver] W3C capabilities and MJSONWP desired capabilities were provided
[debug] [BaseDriver] Creating session with W3C capabilities: {
[debug] [BaseDriver] 	"alwaysMatch": {
[debug] [BaseDriver] 		"platformName": "android",
[debug] [BaseDriver] 		"appium:app": "/Users/siddartha/eclipse-workspace/AppiumSampleDemo/VodQA.apk",
[debug] [BaseDriver] 		"appium:automationName": "UIAutomator2",
[debug] [BaseDriver] 		"appium:deviceName": "android",
[debug] [BaseDriver] 		"appium:platformVersion": "8.1"
[debug] [BaseDriver] 	},
[debug] [BaseDriver] 	"firstMatch": [
[debug] [BaseDriver] 		{}
[debug] [BaseDriver] 	]
[debug] [BaseDriver] }

**Comment by mykola-mokhnach** [comment]
what do you think about `JSON.stringify(w3cCapabilities, 2, ' ')` ?

**Comment by Siddh010** [comment]
Added  third argument with 2, I suppose this look now fine 
log.debug(`Creating session with W3C capabilities: ${JSON.stringify(w3cCapabilities,null,2)}`);

Screen Shot 2019-06-17 at 9 33 02 PM

**Comment by imurchie** [comment]
That option looks good to me.

**Comment by dpgraham** [comment]
Thank you @Siddh010 !

**Comment by Siddh010** [comment]
wow!!   Thanks to every one for guidance and help. @dpgraham, @mykola-mokhnach, @imurchie

**Pull request merged**